tools/print_tock_memory_usage.py

```diff
@@ -0,0 +1,422 @@
+        group_sum = group_sum + group_size
+
+    print(title + ": " + str(group_sum) + " bytes")
+    print(output, end = ' ')
+
+def print_symbol_information():
+    """Print out all of the variable and function groups with their flash/RAM
+       use."""
+    variable_groups = {}
+    group_symbols(variable_groups, kernel_initialized, show_waste, "RAM")
+    group_symbols(variable_groups, kernel_uninitialized, show_waste, "Flash+RAM")
+    if (show_waste):
+        print() # Place an newline after waste reports
+
+    print_groups("Variable groups (RAM)", variable_groups)
+
+    print()
+    print("Embedded data (in flash): " + str(padding_text) + " bytes")
+    print()
+    function_groups = {}
+    # Embedded constants in code (e.g., after functions) aren't counted
+    # in the symbol's size, so detecting waste in code has too many false
+    # positives.
+    group_symbols(function_groups, kernel_functions, False, "Flash")
+    print_groups("Function groups (in flash)", function_groups)
+    print()
+
+def get_addr(symbol_entry):
+    """Helper function for sorting symbols by start address."""
+    return symbol_entry[1]
+
+def compute_padding(symbols):
+    """Calculate how much padding is in a list of symbols by comparing their
+       reporting size with the spacing with the next function and return
+       the total differences."""
+    symbols.sort(key=get_addr)
+    func_count = len(symbols)
+    diff = 0
+    for i in range(1, func_count):
+        (esymbol, eaddr, esize, _) = symbols[i - 1]
+        (_, laddr, _, _) = symbols[i]
+        total_size = laddr - eaddr
+        symbols[i - 1] = (esymbol, eaddr, esize, total_size)
+        if total_size != esize:
+            diff = diff + (total_size - esize)
+
+    return diff
+
+def parse_options(opts):
+    """Parse command line options."""
+    global symbol_depth, verbose, show_waste
+    valid = 'd:vs'
+    long_valid = ['depth=', 'verbose', 'show-waste']
+    optlist, leftover = getopt.getopt(opts, valid, long_valid)
+    for (opt, val) in optlist:
+        if opt == '-d' or opt == '--depth':
+            symbol_depth = int(val)
+        elif opt == '-v' or opt == '--verbose':
+            verbose = True
+        elif opt == '-s' or opt == '--show-waste':
+            show_waste = True
+        else:
+            usage("unrecognized option: " + opt)
+            return []
+
+    return leftover
+
+ # Script starts here ######################################
+arguments = sys.argv[1:]
+if len(arguments) < 1:
+    usage("no ELF specified")
+    sys.exit(-1)
+
+ # The ELF is always the last argument; pull it out, then parse
+ # the others.
+elf_name = ""
+options = arguments
+try:
+    remaining = parse_options(options)
+    if len(remaining) != 1:
+        usage("")
+        sys.exit(-1)
+    else:
+        elf_name = remaining[0]
+except getopt.GetoptError as err:
+    usage(str(err))
+    sys.exit(-1)
+
+header_lines = os.popen('arm-none-eabi-objdump -f ' + elf_name).readlines()
```

Comment:
perhaps extract `arm-none-eabi-objdump` to a variable/constant because it is used twice.

Reply:
Will do and transition to llvm.

Reply:
Done.